### PR TITLE
[WC-910] feat(datagrid-web): double click action

### PR DESCRIPTION
**Pull request by iobuhov**
### Pull request type

New feature (non-breaking change which adds functionality)

---

### Description

Data Grid 1 has a double-click option. This PR adds the double-click option to the DataGrid 2. This will allow developers to replace old data grids with matching functionality.

The double click will be even more effective when the DG2 can emit events for the data view Data source Type "Listen to widget."

An option to consider is to add an extra action for "onClickDouble.". Though this might conflict with the single action, both actions might delay the click action.

### CODE REVIEW

File: packages/pluggableWidgets/datagrid-web/src/helpers/useRowInteractionProps.ts
Comment (samuelreichert) [inline packages/pluggableWidgets/datagrid-web/src/helpers/useRowInteractionProps.ts:1]: Should we change the name of the file? `use-row-interaction-props`?

File: packages/pluggableWidgets/datagrid-web/src/features/keyboard-navigation/useFocusTargetProps.ts
Comment (samuelreichert) [inline packages/pluggableWidgets/datagrid-web/src/features/keyboard-navigation/useFocusTargetProps.ts:1]: Should we change the name of the file? `use-focus-target-props`?

File: packages/pluggableWidgets/datagrid-web/src/features/row-interaction/base.ts [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+    selectionType: SelectionType;
+    clickTrigger: ClickTrigger;
+}
+
+export interface CheckboxContext {
+    item: ObjectItem;
+    pageSize: number;
+    selectionType: SelectionType;
+    selectionMethod: SelectionMethod;
+}
```
[the remark this answers is not in the document]
Comment (iobuhov) [reply packages/pluggableWidgets/datagrid-web/src/features/row-interaction/base.ts]: Done

File: packages/pluggableWidgets/datagrid-web/src/helpers/useRowInteractionProps.ts
Comment (iobuhov) [reply packages/pluggableWidgets/datagrid-web/src/helpers/useRowInteractionProps.ts:1]: This hook is removed

File: packages/pluggableWidgets/datagrid-web/src/features/keyboard-navigation/useFocusTargetProps.ts
Comment (iobuhov) [reply packages/pluggableWidgets/datagrid-web/src/features/keyboard-navigation/useFocusTargetProps.ts:1]: Yes, but I think it's better to do in you pr, wile you moving staff around.

File: packages/pluggableWidgets/datagrid-web/src/helpers/use-select-action-helper.ts [hunk 2]
```diff
@@ -0,0 +1,25 @@
+import { SelectionHelper } from "@mendix/widget-plugin-grid/selection";
```
Comment (gjulivan) [inline packages/pluggableWidgets/datagrid-web/src/helpers/use-select-action-helper.ts:1]: why we need to separate this if this is only a hook wrapper for SelectActionHelper ?
we can combine that on the same file for now with "SelectActionHelper"
unless they serve different helper purpose?

similar with ClickActionHelper and use-click-action-helper.

File: packages/pluggableWidgets/gallery-web/src/features/item-interaction/use-item-events-controller.ts [hunk 3]
```diff
@@ -0,0 +1,15 @@
+import { useMemo } from "react";
```
Comment (gjulivan) [inline packages/pluggableWidgets/gallery-web/src/features/item-interaction/use-item-events-controller.ts:1]: combine file with ItemEventsController ?

**Comment by gjulivan** [comment]
i see here that you started using kebab-case for hooks file naming, is this new file naming convention?
this is seems inconsistent.
don't we have a rule for this?
camelCase for non-react component.
PascalCase for react component.

**Comment by iobuhov** [comment]
> i see here that you started using kebab-case for hooks file naming, is this new file naming convention? this is seems inconsistent. don't we have a rule for this? camelCase for non-react component. PascalCase for react component.

There no rules yet.
My strategy:
PascalCase for classes and components
desh-case for other files

**Comment by gjulivan** [comment]
> > i see here that you started using kebab-case for hooks file naming, is this new file naming convention? this is seems inconsistent. don't we have a rule for this? camelCase for non-react component. PascalCase for react component.
> 
> There no rules yet. My strategy: PascalCase for classes and components desh-case for other files

kebab-case (dash) is already being used for folder naming,
and all of the other hooks file is already using camelCase (similar to react hooks function naming convention)

adding another naming convention here is too much.
why the sudden change?

File: packages/pluggableWidgets/datagrid-web/src/helpers/use-widget-props.ts
Comment (r0b1n) [inline packages/pluggableWidgets/datagrid-web/src/helpers/use-widget-props.ts:1]: This file uses kebab-case

File: packages/pluggableWidgets/datagrid-web/src/features/keyboard-navigation/useFocusTargetProps.ts
Comment (r0b1n) [inline packages/pluggableWidgets/datagrid-web/src/features/keyboard-navigation/useFocusTargetProps.ts:1]: This file uses camelCase. I guess we we lack some consistency.

File: packages/pluggableWidgets/datagrid-web/src/helpers/use-select-action-helper.ts (hunk 2 above)
Comment (iobuhov) [reply packages/pluggableWidgets/datagrid-web/src/helpers/use-select-action-helper.ts:1]: Sure, I'll move it

File: packages/pluggableWidgets/datagrid-web/src/features/keyboard-navigation/useFocusTargetProps.ts
Comment (iobuhov) [reply packages/pluggableWidgets/datagrid-web/src/features/keyboard-navigation/useFocusTargetProps.ts:1]: As we agreed, this will stay camelCasse.

**Review by r0b1n — APPROVED** [review]

**Review by leonardomendix — APPROVED** [review]

**Pull request merged**